cluster.go

```diff
@@ -125,6 +132,33 @@ func (cfg *ClusterConfig) CreateSession() (*Session, error) {
 	return NewSession(*cfg)
 }
 
+// translateHostPort is a helper method that will use the given AddressTranslator
+// if defined, to translate the given host:port (addr) into a new host:port string.
+// If no AddressTranslator or if an error occurs, the given host:port will be returned.
+func (cfg *ClusterConfig) translateHostPort(hostPort string) (string) {
+	if cfg.AddressTranslator == nil {
+		return hostPort
+	}
+	addr, port, err := parseIPPort(hostPort)
+	if err != nil {
+		return hostPort
+	}
+	newAddr, newPort := cfg.AddressTranslator.Translate(addr, port)
+	if gocqlDebug {
+		log.Printf("gocql: translating address '%s' to '%s:%d'", hostPort, newAddr.String(), newPort)
+	}
+	return net.JoinHostPort(newAddr.String(), strconv.Itoa(newPort))
+}
+
+func parseIPPort(ipPort string) (net.IP, int, error) {
+	addr, portStr, err := net.SplitHostPort(ipPort)
+	if err != nil {
+		return nil, 0, err
+	}
+	port, _ := strconv.Atoi(portStr)
+	return net.ParseIP(addr), port, nil
```

Comment:
Yeah, feeding a DNS name in via `NewCluster` will cause the translator to break. My inclination is to do the name resolution here in `translateHostPort`. Since `translateHostPort` is called before dialing we do the DNS lookup at connect time rather than at application start. I feel this is safer and accounts for changes in the DNS record.

Though an argument could be made that the user expects the name to be resolved at the time `NewCluster` is called. I believe doing the name resolution at the time the connection is made to be safer. If the user desires a different functionality he may resolve the names himself prior to calling `NewCluster`.

The chosen approach should be documented for the caller in any case.